src/main/java/ch/njol/skript/conditions/CondIsWithin.java

```diff
@@ -90,54 +98,42 @@ public boolean check(Event event) {
 			return locsToCheck.check(event, box::contains, isNegated());
 		}
 
-		// else, within an entity/block/chunk/world
-		Object area = this.area.getSingle(event);
-		if (area == null)
-			return isNegated();
-
-		// Entities
-		if (area instanceof Entity) {
-			BoundingBox box = ((Entity) area).getBoundingBox();
-			return locsToCheck.check(event, (loc) -> box.contains(loc.toVector()), isNegated());
-		}
-
-		// Blocks
-		if (area instanceof Block) {
-			for (BoundingBox box : ((Block) area).getCollisionShape().getBoundingBoxes()) {
-				// getCollisionShape().getBoundingBoxes() returns a list of bounding boxes relative to the block's position,
-				// so we need to subtract the block position from each location
-				Vector blockVector = ((Block) area).getLocation().toVector();
-				if (!locsToCheck.check(event, (loc) -> box.contains(loc.toVector().subtract(blockVector)), isNegated())) {
+		Object[] areas = area.getArray(event);
+		return locsToCheck.check(event, location ->
+				SimpleExpression.check(areas, object -> {
+					if (object instanceof Entity entity) {
+						BoundingBox entityBox = entity.getBoundingBox();
+						return entityBox.contains(location.toVector());
+					} else if (object instanceof Block block) {
+						// getCollisionShape().getBoundingBoxes() returns a list of bounding boxes relative to the blocks' position,
+						// so we need to subtract the block position from each location.
+						for (BoundingBox blockBox : block.getCollisionShape().getBoundingBoxes()) {
+							Vector blockVector = block.getLocation().toVector();
+							if (!blockBox.contains(location.toVector().subtract(blockVector)))
+								return false;
+						}
+						// if this location is within the block, return true
+						return true;
```

Comment:
I think this logic needs to be swapped around. It should return true if it's in ANY of the bounding boxes, rather than if it's in all of them.